### DISCUSSION TITLE
Implement BoxPlot accounting for a weight column

### DISCUSSION THREAD

**Discussion by caiojp** [vega/altair#3270] [Ideas]
Hello, I was using the mark_boxplot but noticed that it is not able to account for a column with weights for the different lines.

Usually, I work with a lot of data, so we aggregate it by some segments and end up having a column with the number of elements inside each segmentation combination.

I would like to implement a way so the statistical values displayed in the chart account for the count.

![Screenshot 2023-11-19 at 11 38 31](https://github.com/altair-viz/altair/assets/49172908/15fda72a-4d23-4e18-a749-0a6c17364821)
![Screenshot 2023-11-19 at 11 38 47](https://github.com/altair-viz/altair/assets/49172908/4b2a4bde-c57f-4b54-86bb-02f79b0ed87e)

But I don't really know where to start, I am a computer science student and that would be my first contribution to a project.

From what I understand, the [function](https://github.com/altair-viz/altair/blob/main/altair/vegalite/v5/schema/mixins.py#L819) calls for BoxPlotDef inside [core](https://raw.githubusercontent.com/altair-viz/altair/main/altair/vegalite/v5/schema/core.py), but from there I can't find the chart logic.

Can someone help me?

**Comment by binste** [comment]
Altair produces a valid Vega-Lite specification which is then compiled by Vega-Lite to Vega. The actual computation of the values for these box plots happens in Vega (Javascript). See https://altair-viz.github.io/user_guide/internals.html

This feature would need to be implemented in either Vega-Lite or maybe even Vega. I'd suggest that you open a new discussion in the Vega-Lite repo to get inputs from their maintainers if they are interested.

By the way, have you seen https://vegafusion.io/? It can perform calculations, sich as for a box plot, on a server or even a database (duckdb, Snowflake).
